### DISCUSSION TITLE
How can I see the definitions of packages / libs included in Micropython?

### DISCUSSION THREAD

**Discussion by 10bn** [BradenM/micropy-cli#305] [Q&A]
I just started using micropy-cli and created the stubs I needed by using micropy stubs create COM13.
But it doesn't give me the ability jet to go to the definitions of packages that come natively with micropython.
What would I need to do to get this working?

**Comment by BradenM** [comment]
Hi @jaddel,

You bring up a good point.
For non-stdlib dependencies added to a project, this should work "out-of-the-box" since the source code is stored locally alongside the generated stubs. This, however, would not be the case for device-generated stubs / stdlib stubs. 

For now, I would recommend installing the cpython variant of any stdlib you'd like to enable jump to definition for inside your project's virtualenv. This should enable jumping to definition + types. For example, you could `pip install micropython-inspect` in your venv to access the definitions of the [micropython-inspect](https://pypi.org/project/micropython-inspect/) library.

This is a solid candidate for automation, and I will likely add support for it in the future. Thanks!
